src/sinks/kafka.rs

```diff
@@ -303,8 +303,13 @@ impl Sink<Event> for KafkaSink {
         self.seq_head += 1;
 
         let producer = Arc::clone(&self.producer);
+        let kf = self.key_field.clone();
```

Comment:
This is only used as a truth value (`if kf == None` for which, incidentally, clippy may prefer `kf.is_none()`), so it's best to compute that truth value here instead of cloning the underlying value.